core/src/test/scala/org/apache/spark/deploy/ExternalShuffleServiceDbSuite.scala

```diff
@@ -0,0 +1,193 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy
+
+import java.io._
+import java.nio.charset.StandardCharsets
+
+import com.google.common.io.{CharStreams, Closeables, Files}
+import org.scalatest.BeforeAndAfterAll
+
+import org.apache.spark.{SecurityManager, ShuffleSuite, SparkConf, SparkException}
+import org.apache.spark.network.shuffle.{ExternalShuffleBlockHandler, ExternalShuffleBlockResolver}
+import org.apache.spark.network.shuffle.protocol.ExecutorShuffleInfo
+import org.apache.spark.network.util.JavaUtils
+import org.apache.spark.util.Utils
+
+
+/**
+ * This suite gets BlockData when the ExternalShuffleService is restarted
+ * with #spark.shuffle.service.db.enabled = true or false
+ * Note that failures in this suite may arise when#spark.shuffle.service.db.enabled = false
+ */
+class ExternalShuffleServiceDbSuite extends ShuffleSuite with BeforeAndAfterAll {
+  val sortBlock0 = "Hello!"
+  val sortBlock1 = "World!"
+  val SORT_MANAGER = "org.apache.spark.shuffle.sort.SortShuffleManager"
+
+  var sparkConf: SparkConf = _
+  var dataContext: TestShuffleDataContext = _
+
+  var securityManager: SecurityManager = _
+  var externalShuffleService: ExternalShuffleService = _
+  var bockHandler: ExternalShuffleBlockHandler = _
+  var blockResolver: ExternalShuffleBlockResolver = _
+
+  override def beforeAll() {
+    super.beforeAll()
+    sparkConf = new SparkConf()
+    sparkConf.set("spark.shuffle.service.enabled", "true")
+    sparkConf.set("spark.local.dir", System.getProperty("java.io.tmpdir"))
+    Utils.loadDefaultSparkProperties(sparkConf, null)
+    securityManager = new SecurityManager(sparkConf)
+
+    dataContext = new TestShuffleDataContext(2, 5)
+    dataContext.create()
+    // Write some sort data.
+    dataContext.insertSortShuffleData(0, 0,
+      Array[Array[Byte]](sortBlock0.getBytes(StandardCharsets.UTF_8),
+        sortBlock1.getBytes(StandardCharsets.UTF_8)))
+    registerExecutor()
+  }
+
+  override def afterAll() {
+    try {
+      dataContext.cleanup()
+    } finally {
+      super.afterAll()
+    }
+  }
+
+  def registerExecutor(): Unit = {
+    sparkConf.set("spark.shuffle.service.db.enabled", "true")
+    externalShuffleService = new ExternalShuffleService(sparkConf, securityManager)
+
+    // external Shuffle Service start
+    externalShuffleService.start()
+    bockHandler = externalShuffleService.getBlockHandler
+    blockResolver = bockHandler.getBlockResolver
+    blockResolver.registerExecutor("app0", "exec0", dataContext.createExecutorInfo(SORT_MANAGER))
+    blockResolver.closeForTest()
+    // external Shuffle Service stop
+    externalShuffleService.stop()
+  }
+
+  // This test getBlockData will be passed when the external shuffle service is restarted.
+  test("restart External Shuffle Service With InitRegisteredExecutorsDB") {
```

Comment:
I'd reword the test name & comment here a bit.  Maybe for test-name, "Recover shuffle data with spark.shuffle.service.db.enabled=true after shuffle service restart"

and the comment should say something more like "The beforeAll ensures the shuffle data was already written, and then the shuffle service was stopped.  Here we restart the shuffle service and make we can read the shuffle data"

Reply:
done ,thanks your.